hack/custom-patches.sh

```diff
@@ -6,21 +6,31 @@ set -o pipefail
 
 YQ="./bin/yq"
 
-f="helm/cloud-provider-vsphere/charts/vsphere-csi-driver/templates/apps_v1_deployment_vsphere-csi-controller.yaml"
-${YQ} e '.spec.template.spec.securityContext.remove-this-key="'"
+csi_controller_manifest="helm/cloud-provider-vsphere/charts/vsphere-csi-driver/templates/apps_v1_deployment_vsphere-csi-controller.yaml"
+
+# Add SecurityContext to the containers.
+${YQ} eval '.spec.template.spec.securityContext.remove-this-key="'"
 {{- with .Values.podSecurityContext }}
   {{- . | toYaml | nindent 8 }}
 {{- end }}
 "'" | .spec.template.spec.containers[].securityContext.remove-this-key="'"
 {{- with .Values.containerSecurityContext }}
   {{- . | toYaml | nindent 12 }}
 {{- end }}
-"'"' ${f} > ${f}.tmp
-cat ${f}.tmp | grep -v 'remove-this-key' > ${f}
-rm -rf ${f}.tmp
+"'"' ${csi_controller_manifest} > ${csi_controller_manifest}.tmp
 
-# Replace the upstream psp condition with our own.
+# Remove existing runAsNonRoot keys added upstream since we set it in the chart's values.
+# https://github.com/giantswarm/cloud-provider-vsphere-app/blob/6556f98de46ff45b3a8ce9080752ca1050bbee0b/helm/cloud-provider-vsphere/charts/vsphere-csi-driver/values.yaml#L59
+cat ${csi_controller_manifest}.tmp \
+  | grep -v 'remove-this-key' \
+  | grep -v 'runAsNonRoot' \
+  | grep -v 'runAsUser: 65532' \
```

Comment:
I remove the user ID sections to avoid duplicates with the workaround added in the values.